src/rt/lifetime.d

```diff
@@ -1116,7 +1086,44 @@ extern (C) void[] _d_newarraymiT(const TypeInfo ti, size_t ndims, ...)
             va_start(q, __va_argsave);
         else
             static assert(false, "platform not supported");
-        return _d_newarrayOpT!(_d_newarrayiT)(ti, ndims, q);
+
+        auto dims = (cast(size_t*)alloca(size_t.sizeof * ndims))[0..ndims];
+        foreach(ref v; dims)
+            va_arg(q, v);
+        va_end(q);
+        return _d_newarrayOpT!(_d_newarrayiT)(ti, dims);
+    }
+}
+
+
+/**
+ *
+ */
+extern (C) void[] _d_newarraymTX(const TypeInfo ti, size_t[] dims)
+{
+    debug(PRINTF) printf("_d_newarraymT(dims.length = %d)\n", dims.length);
+
+    if (dims.length == 0)
+        return null;
+    else
+    {
+        return _d_newarrayOpT!(_d_newarrayT)(ti, dims);
+    }
+}
+
+
+/**
+ *
+ */
+extern (C) void[] _d_newarraymiTX(const TypeInfo ti, size_t[] dims)
```

Comment:
The way I see it, making the function typesafe variadic would be misleading, as that type of function doesn't actually exist at dmd's elem-level.

When you say currently, do you mean you intend to generate these calls at a higher level inside dmd?